src/app/(protected)/notebooks/page.tsx

```diff
@@ -32,6 +44,31 @@ export default function Page() {
   const user = useAuth();
   const [notebookData, setNotebookData] = useState<Notebook[] | null>(null);
   const [searchData, setSearchData] = useState<Notebook[] | null>(notebookData);
+  const [title, setTitle] = useState("");
+  const [description, setDescription] = useState("");
+
+  const [createNotebook, { data: createNotebookData, loading: createNotebookLoading, error: createNotebookError }] = useMutation(CREATE_NOTEBOOK_MUTATION, {
+    variables: {
+      title: title,
+      description: description,
+      // TO-DO: Current hardcoded userID for testing (need to fix this for dynamic user)
+      ownerId: "68c8ae3b-24d3-4983-b706-41a46b7d5475",
+      ownerType: "user",
+    },
+    onCompleted: () => {
+      Notify.success("Notebook created!");
+    },
+    onError: (error) => {
+      Notify.failure(`${error.message}!`);
+    },
+  });
+
+  useEffect(() => {
+    if (createNotebookData && createNotebookData.createNotebook) {
+      router.push(`/notebooks/${createNotebookData.createNotebook.id}`);
```

Comment:
Instead of using an effect here you can move that to the `onCompleted` callback of the mutation. The callback can take the result data in parameter. Thus, since you need the `id` you need to modify it slightly. 

```ts
onCompleted: (data) => {
      Notify.success("Notebook created!");
      router.push(`/notebooks/${createNotebookData.createNotebook.id}`);
}
```

Now you can remove the `useEffect` completely.